genesyscloud/conversations_messaging_integrations_instagram/resource_genesyscloud_conversations_messaging_integrations_instagram.go

```diff
@@ -0,0 +1,150 @@
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getConversationsMessagingIntegrationsInstagramProxy(sdkConfig)
+
+	log.Printf("Reading conversations messaging integrations instagram %s", d.Id())
+
+	return util.WithRetriesForRead(ctx, d, func() *retry.RetryError {
+		instagramIntegrationRequest, resp, getErr := proxy.getConversationsMessagingIntegrationsInstagramById(ctx, d.Id())
+		if getErr != nil {
+			if util.IsStatus404(resp) {
+				return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read conversations messaging integrations instagram %s: %s", d.Id(), getErr), resp))
+			}
+			return retry.RetryableError(util.BuildWithRetriesApiDiagnosticError(resourceName, fmt.Sprintf("Failed to read conversations messaging integrations instagram %s: %s", d.Id(), getErr), resp))
+		}
+
+		cc := consistency_checker.NewConsistencyCheck(ctx, d, meta, ResourceConversationsMessagingIntegrationsInstagram(), constants.DefaultConsistencyChecks, resourceName)
+
+		resourcedata.SetNillableValue(d, "name", instagramIntegrationRequest.Name)
+
+		if instagramIntegrationRequest.SupportedContent != nil && instagramIntegrationRequest.SupportedContent.Id != nil {
+			_ = d.Set("supported_content_id", *instagramIntegrationRequest.SupportedContent.Id)
+		}
+
+		if instagramIntegrationRequest.MessagingSetting != nil && instagramIntegrationRequest.MessagingSetting.Id != nil {
+			_ = d.Set("messaging_setting_id", *instagramIntegrationRequest.MessagingSetting.Id)
+		}
+
+		resourcedata.SetNillableValue(d, "page_id", instagramIntegrationRequest.PageId)
+		resourcedata.SetNillableValue(d, "app_id", instagramIntegrationRequest.AppId)
+
+		log.Printf("Read conversations messaging integrations instagram %s", d.Id())
+		return cc.CheckState(d)
+	})
+}
+
+// updateConversationsMessagingIntegrationsInstagram is used by the conversations_messaging_integrations_instagram resource to update an conversations messaging integrations instagram in Genesys Cloud
+func updateConversationsMessagingIntegrationsInstagram(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	sdkConfig := meta.(*provider.ProviderMeta).ClientConfig
+	proxy := getConversationsMessagingIntegrationsInstagramProxy(sdkConfig)
+
+	supportedContentId := d.Get("supported_content_id").(string)
+	messagingContentId := d.Get("messaging_setting_id").(string)
+	pageAccessToken := d.Get("page_access_token").(string)
+	userAccessToken := d.Get("user_access_token").(string)
+
+	conversationsMessagingIntegrationsInstagram := platformclientv2.Instagramintegrationupdaterequest{
+		Name:             platformclientv2.String(d.Get("name").(string)),
+		SupportedContent: &platformclientv2.Supportedcontentreference{Id: &supportedContentId},
+		MessagingSetting: &platformclientv2.Messagingsettingrequestreference{Id: &messagingContentId},
+		PageAccessToken:  &pageAccessToken,
+		UserAccessToken:  &userAccessToken,
+	}
```

Comment:
I would extend the function `getConversationsMessagingIntegrationsInstagramFromResourceData` to take a bool which indicates if it is for a create or update, and then return `Instagramintegrationupdaterequest` or `Instagramintegrationrequest` depending on that. At the moment, it's a bit inconsistent looking using it in the create but not for the update